src/common/Command.cpp

```diff
@@ -338,9 +338,12 @@ namespace Cmd {
         while (true) {
             //Check for end of current token
             if (SkipSpaces(in, cmd.end())) {
-                 args.push_back(std::move(currentToken));
```

Comment:
Unfortunately not. Yep, it works in most cases, but still unspecified behavior.

See https://wiki.sei.cmu.edu/confluence/display/cplusplus/EXP63-CPP.+Do+not+rely+on+the+value+of+a+moved-from+object for list of objects in well-specified state after move.  Also string example looks interesting.

Reply:
My reasoning based on standard, which defines that moved-from object is guaranteed to be in a valid state, but explicitly specifies set of STL functions are guaranteed to leave the moved-from object in a well-specified state. 

As I can't find `std::string` functions in this set, it means that moved-from `std::string` is guaranteed to be in a valid state, but exact state is not specified, which leads to unspecified behavior on access to such state. Since unspecified behavior is "behavior, for a well-formed program construct and correct data, that depends on the implementation. The implementation is not required to document which behavior occurs" than it is just implementation details (possibly undocumented) how `std::string` functions will work on moved-from object. 

TLDR; it may work and may not work, depends on implementation.

Reply:
OK, I suppose it's technically correct to say that the state is not fully specified following `clear()`, because the capacity is part of the user-visible state. But we don't do anything that depends on the string's buffer size, so it's fine.

Reply:
Oh, looks reasonable. I'll remove this one.